src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -1687,6 +1674,32 @@ internal sealed override void AddSynthesizedAttributes(PEModuleBuilder moduleBui
                         ImmutableArray.Create(new TypedConstant(compilation.GetWellKnownType(WellKnownType.System_Type), TypedConstantKind.Type, originalType)),
                         isOptionalUse: true));
             }
+
+            return;
+
+            static ArrayBuilder<SynthesizedAttributeData> addPoisonAttributes(ref ArrayBuilder<SynthesizedAttributeData> attributes, CSharpCompilation compilation,
```

Comment:
>ArrayBuilder<SynthesizedAttributeData> [](http://example.com/codeflow?start=19&length=38)

The pattern we use to synthesize attributes is to pass ```ArrayBuilder<SynthesizedAttributeData>``` by ref and return nothing from the method.